Add symmetry property to Orientation-based classes in favour of set_symmetry (#240)
pyxem/orix: CHANGELOG.rst

- Sampling of vectors in UV mesh on a unit sphere (*S2*).
- `ndim` attribute to Object3d and derived classes which returns number of navigation
  dimensions.
- Setting the symmetry of a (Mis)Orientation via a `symmetry.setter`.

Changed
-------
- `StereographicPlot` doesn't use Matplotlib's `transforms` framework anymore, and
  (X, Y) replaces (azimuth, polar) as internal coordinates.
- Renamed `Symmetry` method `fundamental_sector()` to `fundamental_zone()`.

> - `Orientation` class methods `from_euler`, `from_matrix`, and `from_neo_euler` no longer 

Reviewer: Would like to adjust the wording here

  return the smallest angle rotation when `symmetry` argument is provided and not None.
- `CrystalMap.orientations` no longer returns smallest angle rotation.

Deprecated
----------
- The `data_dim` attribute of Object3d and all derived classes is deprecated from 0.8
  and will be removed in 0.9. Use `ndim` instead.
- Setting (Mis)Orientation symmetry via `set_symmetry()` is deprecated in 0.8, in favour of
  setting it directly via a `symmetry.setter`, and will be removed in 0.9. Use

>   `assign_smallest_angle()` instead.

Reviewer 2: ```suggestion
  `map_into_symmetry_reduced_zone()` instead.
```

- Sampling of vectors in UV mesh on a unit sphere (*S2*).
- `ndim` attribute to Object3d and derived classes which returns number of navigation
  dimensions.

> - `symmetry` property for `Orientation` and `Misorientation` classes.

Reviewer 2: Both classes already have a symmetry ([see API](https://orix.readthedocs.io/en/stable/reference.html#orix.quaternion.Orientation.symmetry)).

```suggestion
- Setting the symmetry of a (Mis)Orientation via a `symmetry.setter`.
```

----------
- The `data_dim` attribute of Object3d and all derived classes is deprecated from 0.8
  and will be removed in 0.9. Use `ndim` instead.

> - `set_symmetry` will be deprecated from 0.8 in favour of a `symmetry` property for
>   `Orientation` and `Misorientation` classes. `set_symmetry` will be removed in 0.9.
>   The behaviour of `set_symmetry` can still be achieved by using the following snippet:
>   >>> o = Orientation(data)
>   >>> o.symmetry = symmetry
>   >>> o = o.compute_symmetry_reduced_orientations()

Reviewer 2: We should be brief in the changelog, ideally one, max. two lines per point.

```suggestion
- Setting (Mis)Orientation symmetry via `set_symmetry()` is deprecated in 0.8, in favour of
  setting it directly via a `symmetry.setter`, and will be removed in 0.9. Use
  `assign_smallest_angle()` instead.
```
